scheduler/scheduler.go

```diff
@@ -189,6 +189,25 @@ func (s *Scheduler) Sync(ctx context.Context, client schema.ClientMeta, tables s
 
 	// send migrate messages first
 	for _, table := range tables.FlattenTables() {
+		if syncClient.deterministicCQID {
+			// No PK adjustment should occur if `_cq_id` is not present in the table
+			cqIDCol := table.Columns.Get(schema.CqIDColumn.Name)
+			if cqIDCol == nil || !cqIDCol.PrimaryKey {
+				continue
+			}
+			for i, c := range table.Columns {
+				if !c.PrimaryKey {
+					continue
+				}
+				if c.Name == schema.CqIDColumn.Name {
+					// If CQ_ID is a PK and it should not have a Unique Clause on it
+					table.Columns[i].Unique = false
```

Comment:
`_cq_id` has a unique constraint on it that hinders using deterministic cq id in append mode (where there are no pks).

Reply:
We can do it there as well, but overall why do we want to include the CLI at all when we can do it all in the source and not include the CLI at all?

Reply:
If unique columns are not supported for append mode, we should disable them only for append mode, same as we do for PKs. That way I don't think it will be a breaking schema change (or maybe only breaking if someone's using append mode)

Reply:
Discussed with @hermanschaaf... the `_cq_id` is a reserved column, if it is present it should be considered a PK and we should override it.

Changed in f50a58781e03b04f4c5fa0edaf587f3691cca936

Reply:
Unique constraints on the `_cq_id` do nothing for us in any mode other than adding the amount of data we write...PG destination also can't detect if a unique constraint has changed.

Reply:
OK so looks like we don't consider `unique` here https://github.com/cloudquery/plugin-sdk/blob/a1857c340c77fc55c8e82a129cff81fa7757e5e1/schema/table.go#L458 when detecting changes

Reply:
I still find it confusing that we set `Unique = false` and `PrimaryKey = true`.
It seems the only reason we do it is due to how we implemented append mode, by the CLI removing the primary key constraint and not the unique one. The latter seems like an oversight and we should remove unique from all columns, `_cq_id` or not, when in append mode, otherwise the sync will fail on duplicate values

Reply:
Other than that I went over our destinations and setting `unique` to `false` seems safe when the column is a primary as DBs treat PKs as unique regardless